draft-rbw-add-encrypted-dns-forwarders.md

```diff
@@ -439,6 +440,29 @@ R-SUPPORT-CA: n/a, this system does not use Certification Authorities at all.
 R-SUPPORT-CLIENT: Some; all major libraries support RPK, but clients (browsers and curl) do not support RPK.
 Further, DNR and DDR in verified discovery mode expect to encounter certificates and do not support RPK.
 
+## Self-signed Certificate
+
+A self-signed certificate requires the client to authorize the connection, which is usually
+a "click OK to continue" dialog box and is a trust on first use (TOFU) solution.  While it is
+possible the user verifies the certificate matches expectations, this seldom occurs.
+
```

Comment:
I don't think using self-signed certificate is a good idea. It will not work with IoT devices (without User Interface) supporting Encrypted DNS forwarders.  Most importantly, it trains users to ignore certificate errors and accept certificates which is a poor security practice. Self-signed certificates will only work with DDR in opportunistic mode and it not enabled by default on clients (e.g., windows).

Reply:
@tireddy2, better recoding this as part of the analysis text.

Dan's text does not recommend this and I think is fair.

Reply:
self-signed certificates are included as an example of an existing -- but bad -- practice.  

However, I am unclear how an IoT device, lacking that same user interface, will work with a local CA (where that CA needs to be added to the IoT device's trust list), or work with delegated credentials (should it honor delegated credentials for cpe.danwing.example.com but not cpe.tiru.example.com and how would it know?).

Said another way: do we need a requirement that the solution work, somehow, without user interaction??  I don't know how to achieve that.  

Another point: several modern IoT devices have the ability to get their initial configuration from a smartphone (today, that is mostly just Wi-Fi password), because they lack a user interface. They could expand that capability -- if they cared -- to bootstrap trusting a local (home-installed) CA, do TOFU for DNR, or whatevs.

Reply:
I added a sentence that users become accustomed to the warnings which is bad.

But I'd still like to know if there is need for IoT to get a Requirement, and how we should express that Requirement.

Reply:
IoT may have specifics and may support other DNR variants such as draft-lenders-core-dnr. I tend to keep the discussion in a level that can help adoption of this draft.

The sentence you had about bad practice is good/sufficient for the analysis at this stage. IoT mention if any should be very short. Thanks.

Reply:
Update looks good.